http.py

```diff
@@ -131,7 +133,8 @@ def send_request(self):
             self.sock.send(self.request)
         except socket.error as exc:
             print exc
-        self.get_response()
+        finally:
```

Comment:
Can you move this after `self.sock.send(self.request)? Dont want to call recv on a socket that errors out